src/polygonarea.rs

```diff
@@ -1 +1,202 @@
+////
+/// let (perimeter, area) = pa.compute();
+///
+/// use approx::assert_relative_eq;
+/// assert_relative_eq!(perimeter, 443770.917248302);
+/// assert_relative_eq!(area, 12308778361.469452);
+/// ```
+impl<'a> PolygonArea<'a> {
+    /// Create a new PolygonArea using a Geodesic.
+    pub fn new(geoid: &'a Geodesic) -> PolygonArea {
+        PolygonArea {
+            geoid,
+            num: 0,
+
+            #[cfg(not(feature = "accurate"))]
+            areasum: 0.0,
+
+            #[cfg(feature = "accurate")]
+            areasum: accurate::sum::Sum2::zero(),
+
+            #[cfg(not(feature = "accurate"))]
+            perimetersum: 0.0,
+
+            #[cfg(feature = "accurate")]
+            perimetersum: accurate::sum::Sum2::zero(),
+
+            crossings: 0.0,
+            initial_lat: 0.0,
+            initial_lon: 0.0,
+            latest_lat: 0.0,
+            latest_lon: 0.0,
+        }
+    }
+
+    /// Add a point to the polygon
+    pub fn add_point(&mut self, lat: f64, lon: f64) {
+        if self.num == 0 {
+            self.initial_lat = lat;
+            self.initial_lon = lon;
+        } else {
+            #[allow(non_snake_case)]
+            let (s12, _azi1, _azi2, _m12, _M12, _M21, S12, _a12) =
+                self.geoid
+                    .inverse(self.latest_lat, self.latest_lon, lat, lon);
+            self.perimetersum += s12;
+            self.areasum += S12;
+            self.crossings += PolygonArea::transit(self.latest_lon, lon);
+        }
+        self.latest_lat = lat;
+        self.latest_lon = lon;
+        self.num += 1;
+    }
+
+    /// Add an adge to the polygon using an azimuth (in degrees) and a distance (in meters). This can only be called after at least one point has been added.
+    ///
+    /// # Panics
+    /// Panics if no points have been added yet.
+    pub fn add_edge(&mut self, azimuth: f64, distance: f64) {
+        if self.num == 0 {
+            panic!("PolygonArea::add_edge: No points added yet");
+        }
+
+        #[allow(non_snake_case)]
+        let (lat, lon, _azi2, _m12, _M12, _M21, S12, _a12) =
+            self.geoid
+                .direct(self.latest_lat, self.latest_lon, azimuth, distance);
+        self.perimetersum += distance;
+        self.areasum += S12;
+        self.crossings += PolygonArea::transit(self.latest_lon, lon);
+        self.latest_lat = lat;
+        self.latest_lon = lon;
+        self.num += 1;
+    }
+
+    /// Return perimeter and area of polygon in meters / meters², consuming PolygonArea.
+    pub fn compute(mut self) -> (f64, f64) {
+        #[allow(non_snake_case)]
+        let (s12, _azi1, _azi2, _m12, _M12, _M21, S12, _a12) = self.geoid.inverse(
+            self.latest_lat,
+            self.latest_lon,
+            self.initial_lat,
+            self.initial_lon,
+        );
+        self.perimetersum += s12;
+        self.areasum += S12;
+
+        let perimetersum;
+        let areasum;
+
+        #[cfg(not(feature = "accurate"))]
```

Comment:
Is it possible to do anything useful with PolygonArea without `accurate`? 

If not, maybe it's better to get rid of all the `accurate` feature checks sprinkled around, and add a `polygon_area` feature to this crate which *requires* accurate.

Reply:
(sorry, that was an incomplete thought)

... the idea being that we'd then check for the polygon_area feature just once, in src/lib, and everything here would assume we have it.

Reply:
Sorry, I only just now realized that you're handling the case without `accurate` as well - please disregard my previous two comments. 🤦